test/component/Elsa.Workflows.ComponentTests/Scenarios/WorkflowDefinitionRefresh/DynamicEndpointTests.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Elsa.Workflows.ComponentTests.Helpers.Services;
+using Elsa.Workflows.Runtime.Contracts;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Elsa.Workflows.ComponentTests.Scenarios.WorkflowDefinitionRefresh;
+
+public class DynamicEndpointTests : AppComponentTest
+{
+    private readonly IWorkflowDefinitionsRefresher _workflowDefinitionsRefresher;
+
+    public DynamicEndpointTests(App app) : base(app)
+    {
+        StaticValueHolder.Value = "first-value";
+        _workflowDefinitionsRefresher = Scope.ServiceProvider.GetRequiredService<IWorkflowDefinitionsRefresher>();
+    }
+
+    [Fact]
+    public async Task HelloWorldWorkflow_ShouldRespondWithHelloWorld()
+    {
+        var client = WorkflowServer.CreateHttpWorkflowClient();
+
+        var firstResponse = await client.GetStringAsync("first-value");
+
+        StaticValueHolder.Value = "second-value";
+        await _workflowDefinitionsRefresher.RefreshWorkflowDefinitionsAsync(new Runtime.Requests.RefreshWorkflowDefinitionsRequest
+        {
+            BatchSize = 10,
+            DefinitionIds = ["f69f061159adc3ae"]
+        }, CancellationToken.None);
+
+        var secondResponse = await client.GetStringAsync("second-value");
+
+        Assert.Equal("", firstResponse);
```

Comment:
Why not check against the response code?

Reply:
The test is checking if the new endpoint is set up within the triggers. If this is not the case I believe a 404 should be returned, not a 200.

Reply:
Ah, yes you are right. @MariusVuscanNx what do you think?

Reply:
The method perform an http GET request and tries to retrieve the content (string). If case it doesn't work a 404 will be returned and an exception will be thrown. 

In short it will work only if we have a successful status code.

Reply:
That makes sense. But then, I wonder, if it wouldn't be more reflective of the test to do e.g. `client.SendAsync()`, receive the `HttpResponseMessage` and then explicitly check for its `StatusCode` to be 200.

Reply:
I updated the code to check the status code.